internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -460,6 +485,107 @@ func convertEnvSecurityGroupCfg(mft *manifest.Environment) (*template.SecurityGr
 	}, nil
 }
 
+func (s *LoadBalancedWebService) convertApplicationLoadBalancer() (applicationLoadBalancerConfig, error) {
+	albConfig := s.manifest.RoutingRule
+	if albConfig.Disabled() || albConfig.IsEmpty() {
+		return applicationLoadBalancerConfig{}, nil
+	}
+
+	var aliases []string
+	var err error
+	if s.httpsEnabled {
+		aliases, err = convertAlias(albConfig.Alias)
+		if err != nil {
+			return applicationLoadBalancerConfig{}, err
+		}
+	}
+
+	aliasesFor, err := convertHostedZone(albConfig.Alias, albConfig.HostedZone)
+	if err != nil {
+		return applicationLoadBalancerConfig{}, fmt.Errorf(`convert "http.alias" to string slice: %w`, err)
+	}
+
+	targetContainer, targetPort, err := s.manifest.HTTPLoadBalancerTarget()
+	if err != nil {
+		return applicationLoadBalancerConfig{}, err
+	}
+
+	httpRedirect := true
+	if albConfig.RedirectToHTTPS != nil {
+		httpRedirect = aws.BoolValue(albConfig.RedirectToHTTPS)
+	}
+
+	config := applicationLoadBalancerConfig{
+		settings: &template.ALBListener{
+			Rules: []template.ALBListenerRule{
+				{
+					Path:             aws.StringValue(albConfig.Path),
+					TargetContainer:  targetContainer,
+					TargetPort:       targetPort,
+					Aliases:          aliases,
+					HTTPHealthCheck:  convertHTTPHealthCheck(&albConfig.HealthCheck),
+					Stickiness:       strconv.FormatBool(aws.BoolValue(albConfig.Stickiness)),
+					AllowedSourceIps: convertAllowedSourceIPs(albConfig.AllowedSourceIps),
+					HTTPVersion:      aws.StringValue(convertHTTPVersion(albConfig.ProtocolVersion)),
+				},
+			},
+			RedirectToHTTPS:   httpRedirect,
+			IsHTTPS:           s.httpsEnabled,
+			HostedZoneAliases: aliasesFor,
+		},
+	}
+
+	return config, nil
+}
+
+func (s *BackendService) convertApplicationLoadBalancer() (applicationLoadBalancerConfig, error) {
```

Comment:
I don't quite follow the difference between `*LoadBalancedWebService` method and `*BackendService` they seem pretty similar, can you help me understand why we need to have two separate ones?

Reply:
I wonder if a flow like this would work and reduce duplication:
```go
type loadBalancerTargeter interface {
   HTTPLoadBalancerTarget() (string, string, error)
   MainContainerPort string
}

type routingRuleConfigConverter struct {
   rule manifest.RoutingRuleConfiguration
   manifest loadBalancerTargeter
}   
func (conv routingRuleConfigConverter) convert() (albConfig, error)

type RoutingRuleConfigOrBoolConverter struct {
   rule manifest.RoutingRuleConfigOrBool
   manifest loadBalancerTargeter
}
func (conv routingRuleConfigConverter) convert() (albConfig, error) {
   if conf.rule.Disabled() { return albConfig{}, nil }
   return routingRuleConfigConverter{
        rule: conf.rule.RoutingRuleConfiguration,
        manifest: conv.manifest,
   }.convert()
}
```